### PR TITLE
caloris need & taken chart

### PR DESCRIPTION
**Pull request by vthtuy**

### CODE REVIEW

File: server/routes/profile.js [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
 
+const Tracker = require('../models/tracker');
+
 router.get('/check', async (req, res) => {
   res.json({ success: true })
 });
 
+router.get('/chart/:id', async (req, res) => {
+  var time = Number(req.query.date);  
+  let userId = mongoose.Types.ObjectId(req.params.id);
```
Comment (vnts7) [inline server/routes/profile.js]: userId from req.user._id

File: server/routes/profile.js [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
 });
 
+router.get('/chart/:id', async (req, res) => {
+  var time = Number(req.query.date);  
+  let userId = mongoose.Types.ObjectId(req.params.id); 
+  console.log(time)
+  await Tracker.aggregate([
+      {$match : {userId: userId, date : {$gte : time} }   } 
+      , { "$unwind": "$meals" }
+      , {$group: { "_id": { date: "$date"}, calorieNeeds : {$first : "$calorieNeeds"},  sumTakenCalorie: {$sum: "$meals.calories" } } }
```
Comment (vnts7) [inline server/routes/profile.js]: $sum is not correct
use calorieIntake field

File: server/routes/profile.js [hunk 3]
```diff
@@ -1,9 +1,28 @@
...
+      , {$project : {date : "$_id.date", calorieNeeds : "$calorieNeeds" , sumTakenCalorie : "$sumTakenCalorie" } }
+      , {$sort : {date : 1} }
+    ], 
+    function(err, result) {
```
Comment (vnts7) [inline server/routes/profile.js]: remove this function

File: server/routes/profile.js (hunk 1 above)
Comment (vthtuy) [reply server/routes/profile.js]: /profile/chart/<<userId>> : this is the input.

https://github.com/vnts7/mwa-prj/pull/13/commits/a26e031c064bb230bba9507b1e3044225edc2304  Chart2Component get userId to retrieve data

**Pull request merged**